src/js/modules/RoutingApp.js

```diff
@@ -25,186 +28,174 @@ import {
-                      <EditProfilePage />
-                    ) : (
-                      <Redirect to="/myaccount" />
-                    )}
-                />
-                <Route
-                  exact
-                  path={"/recommended"}
-                  key={"recommended"}
-                  component={RecommendedPage}
-                />
-                <Route
-                  exact
-                  path={"/latest"}
-                  key={"latest"}
-                  component={LatestPage}
-                />
-                <Route
-                  exact
-                  path={"/search"}
-                  key={"search"}
-                  component={SearchPage}
-                />
-                <Route path="*" key={"notFound"} component={NotFoundPage} />
-              </Switch>
-            </PageLayout>
-          )}
-          {initDataError && (
-            <DataErrorPage error={initDataError} action={this.prepareData} />
+      <ConnectedRouter history={appHistory}>
+        <PageLayout>
+          {!loading ? (
+            <Switch>
+              <Route exact path="/" component={HomePage} />
+              {allSections.map(section => {
+                return (
+                  <Route
+                    exact
+                    path={section.permalink}
+                    key={`section${section.id}`}
+                    render={() => <SectionPage section={section} />}
+                  />
+                );
+              })}
+              {allSections.map(section => {
+                return (
+                  <Route
+                    exact
+                    path={`${section.permalink}/:article_slug`}
+                    key={`article${section.id}`}
+                    component={ArticlePage}
+                  />
+                );
+              })}
+              <Route
+                exact
+                path={"/about/:description_slug"}
+                component={DescriptionPage}
+              />
+              <Route
+                exact
+                path={"/contributors/:contributor_slug"}
+                key={"contributors"}
+                component={ContributorPage}
+              />
+              <Route
+                exact
+                path={"/illustrators/:artist_slug"}
+                key={"illustrators"}
+                component={ArtistPage}
+              />
+              <Route
+                exact
+                path={"/photographers/:artist_slug"}
+                key={"photographers"}
+                component={ArtistPage}
+              />
+              <Route
+                exact
+                path={"/recommended"}
+                key={"recommended"}
+                component={RecommendedPage}
+              />
+              <Route
+                exact
+                path={"/latest"}
+                key={"latest"}
+                component={LatestPage}
+              />
+              <Route

```

Comment:
Just a quick comment. This will cause a redirect twice, as it redirects you to signIn no matter what, cause the session data hasn't loaded. Once the session data loads, it redirects you to the account page. This isn't a craaazy issue rn, but if in the future we need to have multiple pages protected by sign in, it will be an issue.